protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -0,0 +1,165 @@
+        return ImmutableList.of(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    protected void doProcess(GetAnnotationRequest message, ImapSession session, String tag, ImapCommand command,
+            Responder responder) {
+        try {
+            proceed(message, session, tag, command, responder);
+        } catch (MailboxNotFoundException e) {
+            session.getLog().info("The command: {} is failed because not found mailbox {}", command.getName(), message.getMailboxName());
+            no(command, tag, responder, HumanReadableText.FAILURE_NO_SUCH_MAILBOX, ResponseCode.tryCreate());
+        } catch (MailboxException e) {
+            session.getLog().info("The command: {} on mailbox {} is failed", command.getName(), message.getMailboxName());
+            no(command, tag, responder, HumanReadableText.GENERIC_FAILURE_DURING_PROCESSING);
+        }
+    }
+
+    private void proceed(GetAnnotationRequest message, ImapSession session, String tag, ImapCommand command, Responder responder) throws MailboxException {
+        String mailboxName = message.getMailboxName();
+        long maxsize = message.getMaxsize();
+        MailboxPath mailboxPath = buildFullPath(session, mailboxName);
+
+        List<MailboxAnnotation> mailboxAnnotations = getMailboxAnnotations(session, message.getKeys(), message.getDepth(), mailboxPath);
+        Optional<Integer> maximumOversizedSize = getMaxSizeValue(mailboxAnnotations, maxsize);
+
+        respond(tag, command, responder, mailboxName, mailboxAnnotations, maxsize, maximumOversizedSize);
+    }
+
+    private void respond(String tag, ImapCommand command, Responder responder, String mailboxName,
+                         List<MailboxAnnotation> mailboxAnnotations, long maxsize, Optional<Integer> maximumOversizedSize) {
+        if (maximumOversizedSize.isPresent()) {
+            responder.respond(new AnnotationResponse(mailboxName, filterItemsBySize(responder, mailboxName, mailboxAnnotations, maxsize)));
+            okComplete(command, tag, ResponseCode.longestMetadataEntry(maximumOversizedSize.get()), responder);
+        } else {
+            responder.respond(new AnnotationResponse(mailboxName, mailboxAnnotations));
+            okComplete(command, tag, responder);
+        }
+    }
+
+    private Optional<Integer> getMaxSizeValue(List<MailboxAnnotation> mailboxAnnotations, long maxsize) {
+        if (maxsize > 0) {
+            return getMaxSizeOfOversizedItems(mailboxAnnotations, maxsize);
+        }
+        return Optional.absent();
+    }
+
+    private List<MailboxAnnotation> filterItemsBySize(Responder responder, String mailboxName, List<MailboxAnnotation> mailboxAnnotations, final long maxsize) {
+        Predicate<MailboxAnnotation> lowerPredicate = new Predicate<MailboxAnnotation>() {
+            @Override
+            public boolean apply(MailboxAnnotation input) {
+                return (input.size() <= maxsize);
```

Comment:
```
return maxSize.transform(new Function<Integer, Boolean>() {
   Boolean apply(Integer maxSize) {
     return (input.size() <= maxSize);
   }).orElse(true);
```
By the way, what happen when size is 0 or -1 (which is default behaviour, I guess?). Nothing is kept ?

Anyway, you also have useless brackets here.